### PR TITLE
ddl, statistics: fix stats meta missing when creating many tables at once

### LINKED ISSUE #36004: fail to notify DDL event and stats_meta is not created for the table
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1. deploy a cluster
2. create a lot of table 

### 2. What did you expect to see? (Required)
1. create table success and no warn log
### 3. What did you see instead (Required)
get the warn log because the notify channel is full.
```log
 [2022/05/25 23:12:18.813 +00:00] [WARN] [ddl.go:255] ["[ddl] fail to notify DDL event"] [event="(Event Type: create table, Table ID: 863784, Table Name archive_blob_2019_07"] 
```
### 4. What is your TiDB version? (Required)
```sql
ySQL root@127.0.0.1:(none)> select tidb_version()\G
***************************[ 1. row ]***************************
tidb_version() | Release Version: v6.1.0-alpha-282-g0c17a9acd
Edition: Community
Git Commit Hash: 0c17a9acdad0b0d6556f37048e9dafb281f58fab
Git Branch: HEAD
UTC Build Time: 2022-07-04 07:32:26
GoVersion: go1.18
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false
```

### LINKED ISSUE #38189: stats meta is not created after creating table
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

Create 2000 tables. Only 797 rows of stats meta are created, which means stats meta of some tables is not created.

### 2. What did you expect to see? (Required)
There should be 2000 rows of stats meta.

### 3. What did you see instead (Required)
Only 797 rows of stats meta are created.

### 4. What is your TiDB version? (Required)
v6.3.0

### PR DESCRIPTION
**Pull request by xuyifangreeneyes**
### What problem does this PR solve?

Issue Number: close #38189 close #36004

Problem Summary:

### What is changed and how it works?

If the stats meta is not inserted after creating the table, we insert it in `DumpStatsDeltaToKV`(use `INSERT INTO ... ON DUPLICATE KEY UPDATE ...`).

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by winoros — APPROVED** [review]

File: statistics/handle/update.go [hunk 1]
```diff
@@ -558,9 +558,13 @@ func (h *Handle) dumpTableStatCountToKV(id int64, delta variable.TableDelta) (up
 			}
 		} else {
 			if delta.Delta < 0 {
-				_, err = exec.ExecuteInternal(ctx, "update mysql.stats_meta set version = %?, count = count - %?, modify_count = modify_count + %? where table_id = %? and count >= %?", startTS, -delta.Delta, delta.Count, id, -delta.Delta)
+				// use INSERT INTO ... ON DUPLICATE KEY UPDATE here to fill missing stats_meta.
+				_, err = exec.ExecuteInternal(ctx, "insert into mysql.stats_meta (version, table_id, modify_count, count) values (%?, %?, %?, 0) on duplicate key "+
+					"update version = values(version), modify_count = modify_count + values(modify_count), count = greatest(count - %?, 0)", startTS, id, delta.Count, -delta.Delta)
```
Comment (xuyifangreeneyes) [inline statistics/handle/update.go:563]: Here we will lose some update for `count` when `count` < `-delta.Delta` or stats meta hasn't been created. Fixing the case needs a more complicated SQL and not fixing it is acceptable. Hence, we choose to not fix the case.

**Review by qw4990 — APPROVED** [review]

File: statistics/handle/update.go (hunk 1 above)
Comment (xuyifangreeneyes) [reply statistics/handle/update.go:563]: ref #5647

**Pull request merged**